extension/main.js

```diff
@@ -55,12 +55,17 @@
 	function handleNotificationClicked(notificationId) {
 		const url = window.GitHubNotify.settings.get(notificationId);
 		if (url) {
-			fetch(url).then(res => res.json()).then(json => {
-				openTab(json.html_url);
+			window.GitHubNotify.request(url).then(res => res.json()).then(json => {
+				if (json.message === 'Not Found') {
+					openTab(window.GitHubNotify.getTabUrl());
+				} else {
+					openTab(json.html_url);
+				}
```

Comment:
`openTab(json.message === 'Not Found' ? window.GitHubNotify.getTabUrl() : json.html_url);`

Shorter still!